atlp-rwanda/Team-furebo-E-commerce-bn
# Ft: Track order #184651353

#### What does this PR do?
his pull request adds functionality to handle the retrieval and updating of order status information by an admin and a buyer
#### Description of Task to be completed?

1. Handle GET requests for order status information.

    - Validate the requesting user/token to be a buyer.
    - Retrieve the appropriate order status information from the database.
    - Send the order status information to the frontend for display to the buyer.
    

2.    Handle POST requests to update order status.

    - Validate the request body.
    - Validate the requesting user/token to be an admin.
    - Update the order status in the database.
    - Send the updated order status to the frontend for display to the buyer.
     
  #### How should this be manually tested?
1. Send a GET request for order status information as a buyer.

    -   Verify that the request is successfully validated.
    -   Ensure that the relevant order status information is retrieved from the database.
    -   Confirm that the order status information is sent to the frontend and displayed correctly.

2. Send a POST request to update the order status as an admin.

    - Ensure that the request body is valid.
    - Verify that the requesting user/token is validated as an admin.
    - Update the order status in the database.
    - Confirm that the updated order status is sent to the frontend and displayed correctly.


#### What are the relevant pivotal tracker/Trello stories?
 #184651353
